plugins/modules/docker_image.py

```diff
@@ -620,16 +623,16 @@ def tag_image(self, name, tag, repository, push=False):
         self.log("image %s was %s" % (repo, found))
 
         if not image or self.force_tag:
-            self.log("tagging %s:%s to %s:%s" % (name, tag, repo, repo_tag))
+            image_name = name
+            if not is_image_name_id(name) and tag and not name.endswith(':' + tag):
+                image_name = "%s:%s" % (name, tag)
+            self.log("tagging %s to %s:%s" % (image_name, repo, repo_tag))
             self.results['changed'] = True
-            self.results['actions'].append("Tagged image %s:%s to %s:%s" % (name, tag, repo, repo_tag))
+            self.results['actions'].append("Tagged image %s to %s:%s" % (image_name, repo, repo_tag))
             if not self.check_mode:
                 try:
                     # Finding the image does not always work, especially running a localhost registry. In those
                     # cases, if we don't set force=True, it errors.
-                    image_name = name
-                    if tag and not re.search(tag, name):
```

Comment:
couldn't leave this comment on a line not included in the PR it seems, but unless I missed any, you've removed all the regex uses, so should be able to remove `import re` I think?

Reply:
Thanks, fixed in dc8550dcb511a3faeebecda40b33a036e5d3f08b.